core/io/http_session.cxx

```diff
@@ -501,6 +501,9 @@ http_session::read_some(
           std::scoped_lock lock(self->current_response_mutex_);
           std::swap(self->current_streaming_response_, ctx);
         }
+        if (ctx.stream_end_handler) {
```

Comment:
could you use the move idiom here?

Reply:
The reason I didn't use it is that we're already moving the whole of `ctx` with `std::swap` and then we just let it get out of scope, it's not used later. Is it still necessary?

Reply:
makes sense, you right